### DISCUSSION TITLE
calculate preferred fps

### DISCUSSION THREAD

**Discussion by 407219125** [FreeRDP/FreeRDP#8657] [General]
why calculate fps by below method:
    encoder->fps = (100 / (inFlightFrames + 1) * encoder->maxFps) /100

**Comment by akallabeth** [comment]
@407219125 you´re talking about `freerdp-shadow`?

it is a (simple) algorithm to reduce the framerate once the client is no longer capable of receiving/decoding the frames in time. (e.g. `inFlightFrames > 0`)
but you´re right, could be written in a simpler fashion ;)
That means:
1. `inFlightFrames = 0 --> encoder->fps = encoder->maxFps`
2. `inFlightFrames = 1 --> encoder->fps = encoder->maxFps / 2` 
3. `inFlightFrames = 2 --> encoder->fps = encoder->maxFps / 3`
4. ...

**Reply by 407219125** [reply]
OK,  I get it, thanks~
